### PR TITLE
[ML] adding some initial document for our pytorch NLP model support

### PR DESCRIPTION
**Pull request by benwtrent**
Adding docs for:

 - put vocab
 - put model definition part
 - start deployment
 - all the new NLP configuration objects for trained model configurations

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 1]
```diff
@@ -443,10 +632,38 @@ An array of input field names for the model.
...
 (Optional, object)
 An object map that contains metadata about the model.
 
+`model_type`::
+(Optional, string)
+The created model type. By default the model type is `tree_ensemble`.
+Appropriate types are: 
++
+--
+* `tree_ensemble`: indicates that the model definition is an ensemble model of decision trees
```
Comment (szabosteve) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:661]:
```suggestion
* `tree_ensemble`: The model definition is an ensemble model of decision trees.
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 2]
```diff
@@ -443,10 +632,38 @@ An array of input field names for the model.
...
+* `lang_ident`: A special type reserved for language identification models
```
Comment (szabosteve) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:662]:
```suggestion
* `lang_ident`: A special type reserved for language identification models.
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 3]
```diff
@@ -912,6 +912,75 @@ Acceptable values are: `string`, `number`, `boolean`. When `boolean` is provided
...
++
+--
+* `[CLS]`: The first token of the sequence being classified.
+* `[SEP]`: Indicates sequence separation. 
+--
+end::inference-config-nlp-tokenization-bert-with-special-tokens[]
+
+tag::inference-config-nlp-tokenization-bert-max-sequence-length[]
+The maximum number of tokens allowed to be output by the tokenizer.
+The default for BERT-style tokenization is 512.
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:939]:
```suggestion
The default for BERT-style tokenization is `512`.
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 4]
```diff
@@ -912,6 +912,75 @@ Acceptable values are: `string`, `number`, `boolean`. When `boolean` is provided
...
+The configuration for retreiving the model's vocabulary. The vocabulary is then
+used at inference time. This information is usually provided automatically by
+storing vocabulary in a known, internally managed index.
+end::inference-config-nlp-vocabulary[]
+
+tag::inference-config-nlp-fill-mask[]
+Configuration for a fill_mask NLP task. The fill_mask task works with models
+optimized for a fill mask action. For example, for BERT models, the following
+text may be provided: "The capital of France is [MASK].". The response indicates
+the value most likely to replace `[MASK]`. In this instance, hopefully the
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:952]:
```suggestion
the value most likely to replace `[MASK]`. In this instance, the
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 5]
```diff
@@ -912,6 +912,75 @@ Acceptable values are: `string`, `number`, `boolean`. When `boolean` is provided
...
+most probable token is `paris`.
+end::inference-config-nlp-fill-mask[]
+
+tag::inference-config-ner[]
+Configures a named entity recognition (NER) task. NER is a special case of token
+classification. Each token in the sequence is classified according to the
+provided classification labels. Currently, the NER task requires the
+`classification_labels` Inside-Outside-Beginning formatted labels. Also, only
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:960]:
```suggestion
`classification_labels` Inside-Outside-Beginning formatted labels. Only
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 6]
```diff
@@ -912,6 +912,75 @@ Acceptable values are: `string`, `number`, `boolean`. When `boolean` is provided
...
+person, organization, location, and miscilaneous are supported.
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:961]:
```suggestion
person, organization, location, and miscellaneous are supported.
```

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 7]
```diff
@@ -168,6 +166,151 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-ner]
++
+.Properties of ner inference
+[%collapsible%open]
+======
+`classification_labels`::::
+(Optional, string) An array of classification labels. NER supports only 
+Inside-Outside-Beginning labels (IOB) and only persons, organizations, locations,
+and miscelanious. For example:
```
Comment (szabosteve) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:226]:
```suggestion
and miscellaneous. For example:
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 8]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+`ner`:::
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-ner]
++
+.Properties of ner inference
+[%collapsible%open]
+=====
+`classification_labels`::::
+(Optional, string) An array of classification labels. NER only supports Inside-Outside-Beginning labels (IOB)
+and only persons, organizations, locations, and miscelanious.
```
Comment (szabosteve) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:462]:
```suggestion
and only persons, organizations, locations, and miscellaneous.
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 9]
```diff
@@ -912,6 +912,75 @@ Acceptable values are: `string`, `number`, `boolean`. When `boolean` is provided
...
+
+tag::inference-config-pass-through[]
+Configures a `pass_through` task. This task is useful for debugging as no
+post-processing is done to the inference output and the raw pooling layer
+results are returned to the caller.
+end::inference-config-pass-through[]
+
+tag::inference-config-text-classification[]
+A text classification task. Text classification classifies a provided text
+sequence into previously known target classes.  A specific example of this is
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:972]:
```suggestion
sequence into previously known target classes. A specific example of this is
```

**Review by szabosteve — COMMENTED** [review]
Thanks for writing this up! 👍  I left a couple of minor suggestions. [One of them](https://github.com/elastic/elasticsearch/pull/78270#discussion_r716706375) – hopefully – will fix the docs-check CI.

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 10]
```diff
@@ -168,6 +166,151 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert]
++
+.Properties of bert
+[%collapsible%open]
+========
+`do_lower_case`::::
+(Optional, boolean)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-do-lower-case]
+
+`max_sequence_length`::::
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:195]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 11]
```diff
@@ -168,6 +166,151 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-max-sequence-length]
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:248]:
```suggestion
(Optional, integer)
include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-max-sequence-length]
```

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 12]
```diff
@@ -168,6 +166,151 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert]
++
+.Properties of bert
+[%collapsible%open]
+========
+`do_lower_case`::::
+(Optional, boolean)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-do-lower-case]
+
+`max_sequence_length`::::
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:293]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 13]
```diff
@@ -183,11 +326,109 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-regression-num
...
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:361]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc [hunk 14]
```diff
@@ -183,11 +326,109 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-regression-num
...
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/get-trained-models.asciidoc:407]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 15]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+
++
+.Properties of bert
+[%collapsible%open]
+=======
+`do_lower_case`::::
+(Optional, boolean)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-do-lower-case]
+
+`max_sequence_length`::::
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:443]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 16]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert]
++
+.Properties of bert
+[%collapsible%open]
+=======
+`do_lower_case`::::
+(Optional, boolean)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-do-lower-case]
+
+`max_sequence_length`::::
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:482]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 17]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:517]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 18]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:575]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc [hunk 19]
```diff
@@ -426,6 +412,209 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
...
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/put-trained-models.asciidoc:609]:
```suggestion
`max_sequence_length`::::
(Optional, integer)
```

File: docs/reference/ml/df-analytics/apis/put-trained-model-vocabulary.asciidoc [hunk 20]
```diff
@@ -39,7 +46,30 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=model-id]
...
 
-////
 [[ml-put-trained-model-vocabulary-example]]
 == {api-examples-title}
-////
+
+The following example shows how to create a model vocabulary for a 
+previously stored trained model configuration.
+
+[source,js]
```
Comment (szabosteve) [inline docs/reference/ml/df-analytics/apis/put-trained-model-vocabulary.asciidoc:55]:
```suggestion
[source,sh]
```

**Review by szabosteve — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**